Formula/netcdf-mpi.rb

```diff
@@ -0,0 +1,153 @@
+  end
+
+  depends_on "cmake" => :build
+  depends_on "gcc"
+  depends_on "hdf5-mpi"
+  depends_on "open-mpi" => [:cc, :cxx, :fc]
+
+  uses_from_macos "curl"
+  conflicts_with "netcdf", because: "netcdf-mpi is a variant of netcdf, one can only use one or the other"
+
+  resource "cxx" do
+    url "https://www.unidata.ucar.edu/downloads/netcdf/ftp/netcdf-cxx4-4.3.1.tar.gz"
+    mirror "https://www.gfd-dennou.org/arch/netcdf/unidata-mirror/netcdf-cxx4-4.3.1.tar.gz"
+    sha256 "6a1189a181eed043b5859e15d5c080c30d0e107406fbb212c8fb9814e90f3445"
+  end
+
+  resource "fortran" do
+    # Source tarball at official domains are missing some configuration files
+    # Switch back at version bump
+    url "https://github.com/Unidata/netcdf-fortran/archive/refs/tags/v4.5.3.tar.gz"
+    sha256 "c6da30c2fe7e4e614c1dff4124e857afbd45355c6798353eccfa60c0702b495a"
+  end
+
+  def install
+    ENV.deparallelize
+    common_args = std_cmake_args << "-DBUILD_TESTING=OFF "
+    # add mpi compilers
+    common_args << "-DCMAKE_C_COMPILER=mpicc"
+    common_args << "-DCMAKE_CXX_COMPILER=mpic++"
+
+    mkdir "build" do
+      args = common_args.dup
+      args << "-DNC_EXTRA_DEPS=-lmpi"
+      args << "-DENABLE_TESTS=OFF" << "-DENABLE_NETCDF_4=ON" << "-DENABLE_DOXYGEN=OFF"
+      # Extra CMake flags for compatibility with hdf5 1.12
+      # Remove with the following PR lands in a release:
+      # https://github.com/Unidata/netcdf-c/pull/1973
+      args << "-DCMAKE_C_FLAGS='-I#{Formula["hdf5-mpi"].include} -DH5_USE_110_API'"
+
+      system "cmake", "..", "-DBUILD_SHARED_LIBS=ON", *args
+      system "make", "install"
+    end
+
+    # Add newly created installation to paths so that binding libraries can
+    # find the core libs.
+    args = common_args.dup << "-DNETCDF_C_LIBRARY=#{lib}/#{shared_library("libnetcdf")}"
+
+    cxx_args = args.dup
+    cxx_args << "-DNCXX_ENABLE_TESTS=OFF"
+    resource("cxx").stage do
+      mkdir "build-cxx" do
+        system "cmake", "..", "-DBUILD_SHARED_LIBS=ON", *cxx_args
+        system "make", "install"
+      end
+    end
+
+    fortran_args = args.dup
+    fortran_args << "-DENABLE_TESTS=OFF"
+    fortran_args << "-DCMAKE_FC_COMPILER=mpifort"
+
+    # Fix for netcdf-fortran with GCC 10, remove with next version
+    ENV.prepend "FFLAGS", "-fallow-argument-mismatch"
+
+    resource("fortran").stage do
+      mkdir "build-fortran" do
+        system "cmake", "..", "-DBUILD_SHARED_LIBS=ON", *fortran_args
+        system "make", "install"
+      end
+    end
+
+    # this was present in the formula for netcdf not sure why
+    # # Remove some shims path
+    # inreplace [
+    #   bin/"nf-config", bin/"ncxx4-config", bin/"nc-config",
+    #   lib/"pkgconfig/netcdf.pc", lib/"pkgconfig/netcdf-fortran.pc",
+    #   lib/"cmake/netCDF/netCDFConfig.cmake",
+    #   lib/"libnetcdf.settings", lib/"libnetcdf-cxx.settings"
+    # ], Superenv.shims_path/ENV.cc, ENV.cc
```

Comment:
If we needed this for `netcdf` we probably still need it.

Reply:
Ah, I see. You skipped the shims by setting `CMAKE_C[XX]_COMPILER`, so this is no longer needed.

Reply:
well I was not sure what was this doing and it was causing errors so I removed it but kept it in case it is still necessary.